provider/spark.go

```diff
@@ -516,28 +626,36 @@ func (e EMRExecutor) InitializeExecutor(store FileStore) error {
 	return store.Write(sparkScriptPath, buff)
 }
 
-func NewSparkExecutor(execType SparkExecutorType, config DatabricksConfig, logger *zap.SugaredLogger) (SparkExecutor, error) {
-	// if execType == EMR {
-	// 	emrConfig := &EMRConfig{}
-	// 	if err := emrConfig.Deserialize(SerializedConfig(config)); err != nil {
-	// 		return nil, fmt.Errorf("Could not deserialize config: %v", err)
-	// 	}
-	// 	client := emr.New(emr.Options{
-	// 		Region:      emrConfig.ClusterRegion,
-	// 		Credentials: aws.NewCredentialsCache(credentials.NewStaticCredentialsProvider(emrConfig.AWSAccessKeyId, emrConfig.AWSSecretKey, "")),
-	// 	})
-
-	// 	emrExecutor := EMRExecutor{
-	// 		client:      client,
-	// 		logger:      logger,
-	// 		clusterName: emrConfig.ClusterName,
-	// 	}
-	// 	return &emrExecutor, nil
-	// } else
-	if execType == Databricks {
-		return NewDatabricksExecutor(config)
+func NewSparkExecutor(execType SparkExecutorType, config SparkExecutorConfig, logger *zap.SugaredLogger) (SparkExecutor, error) {
+	if execType == EMR {
+		emrConfig, ok := config.(*EMRConfig)
```

Comment:
You could consider using a type switch here, then they won't even have to set execType https://go.dev/tour/methods/16